Update rdf_to_excel to work with template 0.4.3

dalito opened the pull request:
Some notes:

- I had to deactivate validation. (but didn't look in detail why)
- the blank template could be cleaned up a bit more (reset zoom level, unhide cols/rows etc.)
- pillow is required to write Excel files with images (without pillow all images are removed without notice)
- concepts in different languages are reported with one row per language
- requirement-files and poetry.lock not yet updated

Instead of hard-coding the blank template it would be nice to add an option to use another (branded) template-file. This an be done later in a separate PR.

peterphilips commented:
hey @dalito , can you help explain to me what's going on here: https://github.com/surroundaustralia/VocExcel/pull/30/commits/3de62a9c03d414ba2ef1a1ef0a63ae3ef7745202

dalito commented:
Ok. I'll try.

I used tests/eg-valid.ttl for testing plus a multi-language ttl I created. I had to deactivate validation by pashacl because it fails for `tests/eg-valid.ttl` (should probably get its own issue).
```
The file you supplied is not valid according to the vocpub profile.
Traceback (most recent call last):
  File "C:\Users\dlinke\MyProg_exp040\VocExcel\vocexcel\convert.py", line 579, in main
    o = rdf_to_excel(
  File "C:\Users\dlinke\MyProg_exp040\VocExcel\vocexcel\convert.py", line 252, in rdf_to_excel
    raise ConversionError(
utils.ConversionError: The file you supplied is not valid according to the vocpub profile.
```
The main changes were driven by the changes made to the new 0.4.3 template: collections have to their own sheet, changes to columns, changes to position. 

I recognized that some labels were not made to support several languages and it seems that output of multiple languages to Excel was not supported before. So
 - in `class Concept(BaseModel): `pref_label` and `definition` were changed to list type. 
 - `holder` in rdf_to_excel had to be changed accordingly
 -  Because language output was not So I decided to write on line per language but do add "Children URI" and "Source Vocab URI" not repeated in every line but just in the first line (which is "en" if present). If there is more than one language row-count does not increase by concept but by number of languages in a concept which is handle by `row_no_concepts`. 
 - Note: alt_labels are currently supporting only one language: Changing this requires a template change.

Regarding language support: Currently languages can be mixed in a single row. You can add a preferred label in Italian and a definition in Chinese in the same row. I think this adds no value. Instead, having one language per line reflects how translations happen and it can be immediately seen if e.g. a definition was not yet translated.

@peterphilips Please add comment/questions directly in the PR-code if you want me to explain some specific details.

dalito commented:
Can you point me to a test that could be updated/reused for testing this? Should testing be added as part of this PR or as a follow-up?

peterphilips commented:
I see - okay let me tinker around with the pyshacl and get back to you regarding problems I encounter

peterphilips commented:
> Can you point me to a test that could be updated/reused for testing this? Should testing be added as part of this PR or as a follow-up?

I'll have to evaluate this. Either or however regarding implementation of tests. 

At a first glance however, this doesn't seem to change the rdf output. I doubt this will cause issues in test cases but I will see!

dalito commented:
@peterphilips - I think this PR is much better now. I have changed the Concept-model to accept for `pref_label` and for `definition` both `str` (as before) and `List[str]` (to account for multiple languages). Now all tests pass. 

Apparently I had forgotten to run the test before I created the PR. Sorry If this has caused confusion. I have added a CI-config to prevent this from happening again.

dalito commented:
I removed CI from this PR.

dalito commented:
This should be good to merge now. I reactivated pyshacl validation in `rdf_to_excel` and all fixed remaining problems (altLabel, custodian and pid were lost before in the conversion).

Now it is possible to repeatedly convert excel to turtle and all features in the template are supported. When the conversion is done multiple times or with a differently sorted Excel vocabularies, the turtle files are always identical.

peterphilips commented:
Apologies @dalito, I was recently very sick. These changes are incredibly!

dalito commented:
@peterphilips - don`t worry! I wish you all the best for a full and speedy recovery. This was kind of fundamental to a establish a full workflow. Great that you merged it.
